boto3/examples/s3.rst

```diff
@@ -0,0 +1,15 @@
+**To list objects in an Amazon S3 bucket**
```

Comment:
I wonder if it would be cool if we had a table of contents for the table section with links to each section like we do when documenting client and resource methods if the examples were ever to get too large.